src/studio/src/designer/frontend/packages/schema-editor/src/features/editor/schemaEditorSlice.ts

```diff
@@ -242,11 +242,19 @@ const schemaEditorSlice = createSlice({
       const {
         path, oldKey, newKey,
       } = action.payload;
+      if (oldKey === newKey) {
+        return;
+      }
+      let key = newKey;
       const schemaItem = getUiSchemaItem(state.uiSchema, path);
       if (schemaItem.restrictions) {
+        // eslint-disable-next-line no-loop-func
```

Comment:
What kind of key are we trying to set here?
where is it used?

Reply:
The restriciton key name, ex minLength, maxLength etc.

Reply:
renamed to setRestrictionKey

Reply:
Could you add a comment on why you are ignoring the linter?
why are the keys is the fieldItem.key a number value?
if that is the case, why not use a number-store that iterates every time it assigns a integer?
That way you do not need to have a loop, and all values are unique.

Reply:
Im simply adding a 1 to the end of the keyname, to prevent duplicates. It was a quick fix, probably better ways.

Reply:
This is also a place you can use the getUniqueNumber function, as you would only need to check for one duplicate, the number trailing would always be unused